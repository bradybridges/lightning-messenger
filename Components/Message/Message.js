import React, { Component } from 'react';
import { Text, View, StyleSheet, Dimensions } from 'react-native';

export default class Message extends Component {
  formatTimestamp = (timestamp) => {
    const date = new Date(timestamp.seconds * 1000);
    let hours = Number(date.getHours());
    const label = (hours >= 12) ? 'PM': 'AM';
    let minutes = Number(date.getMinutes());
    if(hours > 12) {
      hours = hours - 12;
    }
    if(minutes < 10) {
      minutes = `0${minutes}`;
    }
    return `${hours}:${minutes} ${label}`;
  }
  render() {
    const { content, timestamp } = this.props;
    const time = this.formatTimestamp(timestamp);
    return (
      <View style={styles.container}>
        <Text style={styles.content}>{content}</Text>
        <Text style={styles.timestamp}>{time}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: 'red',
    borderRadius: 100,
    marginVertical: 30,
  },
  content: {
    padding: 10,
    fontSize: 18,
  },
  timestamp: {
    alignSelf: 'flex-end',
    marginRight: 20,
  }
});

