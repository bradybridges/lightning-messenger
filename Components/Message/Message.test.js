import React from 'react';
import { shallow } from 'enzyme';
import Message from './Message';

describe('Message', () => {
  let wrapper;
  const mockDate = { seconds: '1581565040' };

  beforeEach(() => {
    wrapper = shallow(
      <Message
        content="test message"
        timestamp={mockDate}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('formatTimestamp should return the time of a timestamp with correct AM and PM labels', () => {
    const expectedTime = '8:37 PM';
    const time = wrapper.instance().formatTimestamp(mockDate);
    expect(time).toEqual(expectedTime);
  });
});
