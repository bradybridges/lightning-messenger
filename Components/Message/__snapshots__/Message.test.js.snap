// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#6E8F76",
      "borderColor": "#abababeb",
      "borderRadius": 25,
      "borderWidth": 1,
      "display": "flex",
      "marginLeft": 10,
      "marginTop": 15,
      "paddingBottom": 5,
      "width": 675,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 18,
        "padding": 5,
      }
    }
  >
    test message
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "flex-end",
        "color": "black",
        "fontSize": 12,
        "marginRight": 20,
      }
    }
  >
    8:37 PM
  </Text>
</View>
`;
